src/runtime_src/core/common/api/xrt_device.cpp

```diff
@@ -241,6 +242,12 @@ boost::any
 device::
 get_info(info::device param) const
 {
+  auto json_str = [](boost::property_tree::ptree pt) {
+    std::stringstream ss;
+    boost::property_tree::write_json(ss, pt);
+    return ss.str();
+  };
```

Comment:
Please move as static function under existing namespace query within this file.  I imagine the function argument should be `const boost::property_type::ptree& pt`

Reply:
Moved. I pass in the ptree directly so it can't be passed in as a reference. Usage:
```
case info::device::thermals :               // std::string
    return query::json_str(xrt_core::sensor::read_thermals(handle.get()));
```
Should I change it to:
```
case info::device::thermals :               // std::string
    auto thermal = xrt_core::sensor::read_thermals(handle.get());
    return query::json_str(thermal);
```
If so, can you help me understand why the 2nd one is better?

Reply:
First is best, I didn't realize this.

Reply:
You should use the first, but still use const type& in declaration.